### PR TITLE
HL7 base64 fix

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- modified code which support base and soft lab result
- added date util package

### CODE REVIEW

File: integration-pipeline/HL7v2/resources/diagnosticreport.py [hunk 1]
```diff
@@ -87,13 +88,13 @@ def prepare_diagnostic_report(data, patient: Patient, encounter: Encounter, pare
...
             id=get_md5(),
             practitioner=Reference(
-                extension=[Extension(
-                    url="http://hl7.org/fhir/StructureDefinition/event-performerFunction",
-                    valueCodeableConcept=CodeableConcept(coding=[Coding(
-                        system="http://terminology.hl7.org/CodeSystem/v3-ParticipationType",
-                        code="SPRF"
-                    )])
-                )],
+                # extension=[Extension(
```
Comment (tamsud) [inline integration-pipeline/HL7v2/resources/diagnosticreport.py:91]: why the comment is added here

File: integration-pipeline/HL7v2/resources/observation.py [hunk 2]
```diff
@@ -127,14 +132,14 @@ def prepare_observation(
...
 
-    if "ED" in data.get("value", {}):
-        observation.extension = list(map(lambda attachment_data: Extension(
-            url="https://hl7.org/fhir/R5/StructureDefinition/extension-Observation.valueAttachment",
-            valueAttachment=Attachment(
-                data=attachment_data.get("data"),
-                contentType=attachment_data.get("data_subtype")
-            )
-        ), data.get("value", {}).get("ED", {})))
+    # if "ED" in data.get("value", {}):
```
Comment (tamsud) [inline integration-pipeline/HL7v2/resources/observation.py:135]: why the code is commented out here

File: integration-pipeline/HL7v2/resources/observation.py [hunk 3]
```diff
@@ -214,4 +219,13 @@ def prepare_observation(
     if encounter is not None:
         observation.encounter = Reference(reference="Encounter/" + encounter.id)
 
+    if 'value' in data and 'type' in data['value'] and data['value']['type'] == "ED":
+        pdf_data = base64.b64decode(data['value']['ED'][0]['data'])
+        upload_pdf = BytesIO(pdf_data)
+        CONNECTION_STRING = os.environ.get("CONNECTION_STRING")
+        upload_blob = HL7Processor.upload_to_azure_blob(container_name=CONTAINER_NAME,
```
Comment (tamsud) [inline integration-pipeline/HL7v2/resources/observation.py:226]: extension of the file is missing

Comment (tamsud) [reply integration-pipeline/HL7v2/resources/observation.py:226]: blob_name=f"{patient.id}/{observation.id}",

File: utils/base64.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    @staticmethod
+    def save_base64_to_pdf(base64_str, output_path):
+        """
+        Save a base64 encoded string as a PDF file.
+
+        Args:
+            base64_str (str): The base64 encoded string.
+            output_path (str): The path where the PDF should be saved.
+        """
+        with open(output_path, "wb") as pdf_file:
```
Comment (tamsud) [inline utils/base64.py:35]: when the saved file be deleted

**Review by tamsud — CHANGES_REQUESTED** [review]

File: integration-pipeline/HL7v2/resources/diagnosticreport.py (hunk 1 above)
Comment (MonishaSivanathan) [reply integration-pipeline/HL7v2/resources/diagnosticreport.py]: removed the comment

File: integration-pipeline/HL7v2/resources/observation.py (hunk 3 above)
Comment (MonishaSivanathan) [reply integration-pipeline/HL7v2/resources/observation.py]: added the extention

File: utils/base64.py (hunk 4 above)
Comment (MonishaSivanathan) [reply utils/base64.py]: for testing purpose added the functiion, removed the file.

**Review by tamsud — APPROVED** [review]

**Pull request merged**